SalaryService.Application/Services/CompensationsService.cs

```diff
@@ -0,0 +1,54 @@
+﻿using SalaryService.Application.Commands;
+using SalaryService.Application.Dtos;
+using SalaryService.Application.Queries.Contracts;
+using SalaryService.Domain;
+
+namespace SalaryService.Application.Services;
+
+public class CompensationsService
+{
+    private readonly CompensationCreationCommand _compensationCreationCommand;
+
+    private readonly ICompensationsQuery _compensationsQuery;
+
+    public CompensationsService(CompensationCreationCommand createCompensationCommandHandler, ICompensationsQuery compensationsQuery)
+    {
+        _compensationCreationCommand = createCompensationCommandHandler;
+        _compensationsQuery = compensationsQuery;
+    }
+
+    public async Task<List<CompensationType>> GetTypesAsync()
+    {
+        return CompensationTypes.GetTypeList();
+    }
+
+    public async Task<CompensationListDto> GetAllAsync()
+    {
+        var compensations = await _compensationsQuery.GetCompensationsAsync();
+
+        var compensationList = compensations.Select(x => new CompensationItemDto()
+        {
+            Id = x.Id,
+            Comment = x.Comment,
+            Amount = x.Amount,
+            IsPaid = x.IsPaid,
+            DateCreateCompensation = x.DateCreateCompensation.ToString(),
+            DateCompensation = x.DateCompensation.ToString()
+        }).ToList();
+
+        var totalUnpaidAmount = compensations.Sum(x => x.Amount);
+
+        var compensationsResponseList = new CompensationListDto()
+        {
+            List = compensationList,
+            TotalUnpaidAmount = totalUnpaidAmount
+        };
```

Comment:
all this you can do in the DTO model. It will look better

Reply:
i'll do it later